### PR TITLE
adding new job notify-slack 

### PR DESCRIPTION
**Pull request by SubhedarV**
- added new job notify-slack to get notification on slack channel on failure of any job for master pipeline

### CODE REVIEW

File: .github/workflows/xrt_master.yml [hunk 1]
```diff
@@ -591,5 +591,16 @@ jobs:
           github-server-url: ${{ secrets.SERVER_URL }}
           sshKey: ${{ secrets.CI_PRIVATE_SSH_KEY }}
           runnerName: ${{ runner.name }}
-  
 
+  notify-slack:  
+    needs: [build, windows-build, apu-package-build, package-download, extract-platforms, pcie-hw-tests, pcie-sw-emulation-tests, pcie-hw-emulation-tests, test-summary, setenv-download, Release_to_xco, Release_to_all_sites, Upload_packages_to_Artifactory, gradle-artifactory-upload]
```
Comment (manikandan-xilinx) [inline .github/workflows/xrt_master.yml:596]: Do we need to add all jobs name? can't we just use needs "gradle-artifactory-upload" ?

Comment (SubhedarV) [reply .github/workflows/xrt_master.yml:596]: Let me test that

Comment (manikandan-xilinx) [reply .github/workflows/xrt_master.yml:596]: actually we would need "Release_to_all_sites" and "gradle-artifactory-upload"

Comment (SubhedarV) [reply .github/workflows/xrt_master.yml]: Yes, now I added dependency on "Release_to_all_sites" and "gradle-artifactory-upload"

**Review by manikandan-xilinx — APPROVED** [review]

**Pull request merged**